docs/asl/tutorial.md

```diff
@@ -1,28 +1,220 @@
+This example pushes two strings `nf.cluster` and `discovery` onto the stack and then executes the
+command `:eq`. The equal command pops two strings from the stack and pushes a query object onto the
+stack. The behavior can be described by the stack effect `String:key String:value – Query`. We then
+push a list of tag keys to the stack and execute the command `:by` to group the results.
 
-### Atlas Stack Language
+## Parts
 
-## Basic Operators
+There are only four reserved symbols used for structuring the expression: `,:()`
 
-### Choosing
+1. Commas separate items on the stack. So `a,b` puts two strings on the stack with values `"a"`
+   and `"b"`.
+2. Colon is used to prefix operations. If the first character is a colon the item will be treated
+   as a command to run. For example, `a,:dup`, will push `"a"` on the stack and then execute the
+   [duplicate](std-dup) operation.
+3. Parenthesis are used to indicate the start and end of a list. The expression `(,)` puts an
+   empty list on the stack. Commands inside of a list will not be executed unless the list is
+   passed to the [call](std-call) command. For example, `(,:dup,)` will push a list with a single
+   string value of `":dup"` on to the stack.
+
+## Data Model
+
+The stack language is primarily used for representing expressions over tagged time series
+data. A tag is a string key value pair used to describe a measurement. Atlas requires at least
+one tag with a key of `name`. Example tags represented as a JSON map:
+
+```json
+{
+  "name":       "jvm.gc.pause",
+  "cause":      "Allocation_Failure",
+  "statistic":  "count",
+  "nf.app":     "www",
+  "nf.cluster": "www-main",
+  "nf.asg":     "www-main-v001",
+  "nf.stack":   "main",
+  "nf.node":    "i-01",
+  "nf.region":  "us-east-1",
+  "nf.zone":    "us-east-1a"
+}
+```
+
+Typically, tags should be dimensions that allow you to use the name as a pivot and other tags
+to drill down into the data. The tag keys are similar to columns in a traditional table, however,
+it is important to note that not all time series will have the same set of tag keys.
+
+The tags are used to identify a time series, which conceptually is a set of timestamp value pairs.
+Here is a simplified data set shown as a table:
+
+| name        | app   | node   | values                         |
+|-------------|-------|--------|--------------------------------|
+| cpuUsage    | www   | i-01   | [(05:00, 33.0), (05:01, 31.0)] |
+| cpuUsage    | www   | i-02   | [(05:00, 20.0), (05:01, 37.0)] |
+| cpuUsage    | db    | i-03   | [(05:00, 57.0), (05:01, 62.0)] |
+| diskUsage   | www   | i-01   | [(05:00,  9.0), (05:01,  9.0)] |
+| diskUsage   | www   | i-02   | [(05:00,  7.0), (05:01,  8.0)] |
+| requestRate | www   |        | [(05:00, 33.0), (05:01, 31.0)] |
+
+The table above will be used for the examples in later sections.
+
+## Simple Expressions
+
+All expressions generally have four parts:
+
+1. **Query:** selects a set of time series.
```

Comment:
We typically refer to this as choosing. Internally it is the Query type, but query is often used in conversation to mean the entire expression which gets confusing. Choosing also gives the CAMP acronym to serve as a reminder.

Reply:
Ack, I can change it elsewhere too. How about `Predicate` and changing to `PAMP`? ;)